worker/include/RTC/SCTP/association/Socket.hpp

```diff
@@ -17,82 +21,128 @@ namespace RTC
-				/**
-				 * Maximum received window buffer size. It must be larger than the
-				 * largest sized message we want to be able to receive.
-				 *
-				 * @remarks
-				 * Default value copied from dcSCTP library.
-				 */
-				uint32_t myAdvertisedReceiverWindowCredit{ 5 * 1024 * 1024 };
-				/**
-				 * Use Partial Reliability Extension.
-				 * @see RFC 3758.
-				 */
-				bool partialReliability{ false };
-				/**
-				 * Use Stream Schedulers and User Message Interleaving (I-DATA Chunks).
-				 * @see RFC 8260.
-				 */
-				bool messageInterleaving{ false };
-				/**
-				 * Alternate error detection method for Zero Checksum.
-				 *
-				 * @remarks
-				 * This feature is only enabled if both peers signal their wish to use
-				 * the same (non-zero) Zero Checksum Alternate Error Detection Method.
-				 *
-				 * @see RFC 9653.
-				 */
-				ZeroChecksumAcceptableParameter::AlternateErrorDetectionMethod zeroCheksumAlternateErrorDetectionMethod{
-					ZeroChecksumAcceptableParameter::AlternateErrorDetectionMethod::NONE
-				};
-				/**
-				 * Maximum size of a SCTP packet. It doesn't include any overhead of
-				 * DTLS, TURN, UDP or IP headers.
-				 */
-				size_t mtu{ RTC::Consts::MaxSafeMtuSizeForSctp };
+			public:
+				virtual ~Listener() = default;
+
+			public:
+				virtual void OnSocketSendSctpPacket(const Socket* socket, Packet* packet) const = 0;
+			};
+
+		public:
+			/**
+			 * SCTP association state.
+			 */
+			enum class State
+			{
+				CLOSED,
+				COOKIE_WAIT,
+				// NOTE: TCB is valid in these states:
+				COOKIE_ECHOED,
+				ESTABLISHED,
+				SHUTDOWN_PENDING,
+				SHUTDOWN_SENT,
+				SHUTDOWN_RECEIVED,
+				SHUTDOWN_ACK_SENT,
+			};
+
+			/**
+			 * Struct holding local verification tag and initial TSN between having
+			 * sent the INIT Chunk until the connection is established (there is no
+			 * TCB in between).
+			 *
+			 * @remarks
+			 * This is how dcSCTP does, despite RFC 9260 states that the TCB should
+			 * also be created when an INIT Chunk is sent.
+			 */
+			struct PreTransmissionControlBlock
+			{
+				uint32_t myVerificationTag{ 0 };
+				uint32_t myInitialTsn{ 0 };
 			};
 
 		public:
-			explicit Socket(SocketOptions options);
+			static constexpr std::string_view State2String(State state);
+
+		public:
+			explicit Socket(SocketOptions options, Listener* listener);
 
 			~Socket();
 
 			void Dump(int indentation = 0) const;
 
+			/**
+			 * Initiate the SCTP association with the remote peer. It sends an INIT
+			 * Chunk.
+			 *
+			 * @remarks
+			 * The Socket must be in Closed state.
+			 */
+			void Associate();
+
 		private:
-			void SendInitChunk();
+			void SetState(State state, const std::string& reason);
+
+			Packet* CreatePacket() const;
+
+			Packet* CreatePacketWithPeerVerificationTag(uint32_t peerVerificationTag) const;
+
+			void AddCapabilitiesParametersToChunk(Chunk* chunk) const;
+
+			void CreateTransmissionControlBlock(
+			  uint32_t myVerificationTag,
```

Comment:
Probably nonsense, but doesn't the typical `local/remote` naming work here rather than `my/peer`?

Reply:
Ok, done.